internal/pkg/v2/infrastructure/redis/device_service.go

```diff
@@ -0,0 +1,68 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package redis
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/edgexfoundry/edgex-go/internal/pkg/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/errors"
+	model "github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+
+	"github.com/gomodule/redigo/redis"
+)
+
+const DeviceServiceCollection = "v2:deviceService"
+
+// addDeviceService adds a new device service into DB
+func addDeviceService(conn redis.Conn, ds model.DeviceService) (addedDeviceService model.DeviceService, edgeXerr errors.EdgeX) {
+	// retrieve Device Service by Id first to ensure there is no Id conflict; when Id exists, return duplicate error
+	_, edgeXerr = deviceServiceById(conn, ds.Id)
+	if errors.Kind(edgeXerr) != errors.KindEntityDoesNotExist {
+		return addedDeviceService, errors.NewCommonEdgeX(errors.KindDuplicateName, fmt.Sprintf("device service id %s already exists", ds.Id), nil)
+	}
+
```

Comment:
edgeXerr should be assigned to nil, if it is KindEntityDoesNotExist

Reply:
Added an else condition to set `edgeXerr` back to `nil`.